### PR TITLE
Feat/swift5

### PR DESCRIPTION
**Pull request by cdeust**
Thanks for proposing a pull request.

To help us review the request, please complete the following:

- [X] sign [contributor license agreement](https://developers.facebook.com/opensource/cla)
- [X] describe the change (for example, what happens before the change, and after the change)

Changes:

Making project compliant with Swift5 Compiler
Also made the modification needed to make it working with Facebook-swift-sdk using Swift5 Compiler.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Project Owners
+<<<<<<< HEAD
```
Comment (codytwinton) [inline .github/CODEOWNERS]: This appears to be a merge conflict committed.

Comment (cdeust) [reply .github/CODEOWNERS]: Yeah I figured there is some merge conflicts which were commited and that should not, I'll make an update of the merge request after rechecking all the files and fixing the conflicts not handled

File: FBSDKCoreKit/FBSDKCoreKit/AppLink/FBSDKAppLinkResolver.m [hunk 2]
```diff
@@ -79,7 +79,7 @@ - (void)appLinksFromURLs:(NSArray<NSURL *> *)urls handler:(FBSDKAppLinksBlock)ha
   NSMutableDictionary<NSURL *, FBSDKAppLink *> *appLinks = [NSMutableDictionary dictionary];
   NSMutableArray<NSURL *> *toFind = [NSMutableArray array];
   NSMutableArray<NSString *> *toFindStrings = [NSMutableArray array];
-
+
```
Comment (codytwinton) [inline FBSDKCoreKit/FBSDKCoreKit/AppLink/FBSDKAppLinkResolver.m]: Needs whitespace trimming

**Comment by codytwinton** [comment]
I know understand the goal behind this PR. All of these changes can/should be able to be found on the [`4.x-branch`](https://github.com/facebook/facebook-objc-sdk/tree/4.x-branch).

**Comment by cdeust** [comment]
I'm not sure though the 4.x branch will be working if you don't update projects settings and comply with Swift5 Compiler assuming you want to have the sdk compiled using the latest Swift Compiler of course.

**Pull request closed without merging**